src/System.Devices.Gpio/System.Devices.Gpio/GpioExtensions.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace System.Devices.Gpio
+{
+    public static class GpioExtensions
+    {
+        public static void Set(this GpioPin pin)
+        {
+            pin.Write(GpioPinValue.High);
+        }
+
+        public static void Clear(this GpioPin pin)
+        {
+            pin.Write(GpioPinValue.Low);
+        }
+
+        public static void Toogle(this GpioPin pin)
+        {
+            GpioPinValue value = pin.Read();
+
+            switch (value)
+            {
+                case GpioPinValue.Low: value = GpioPinValue.High; break;
+                case GpioPinValue.High: value = GpioPinValue.Low; break;
```

Comment:
nit: not the focus at this moment but the enum can be cast from an integer and the API likely should throw if the value is not expected